module/localisation/RobotLocalisation/src/RobotLocalisation.cpp

```diff
@@ -96,19 +111,52 @@ namespace module::localisation {
 
                 // **Emit the localisation of the robots**
                 auto localisation_robots = std::make_unique<LocalisationRobots>();
-                for (const auto& tracked_robot : tracked_robots) {
-                    auto state = RobotModel<double>::StateVec(tracked_robot.ukf.get_state());
-                    LocalisationRobot localisation_robot;
-                    localisation_robot.id                  = tracked_robot.id;
-                    localisation_robot.rRWw                = Eigen::Vector3d(state.rRWw.x(), state.rRWw.y(), 0);
-                    localisation_robot.vRw                 = Eigen::Vector3d(state.vRw.x(), state.vRw.y(), 0);
-                    localisation_robot.covariance          = tracked_robot.ukf.get_covariance();
-                    localisation_robot.time_of_measurement = tracked_robot.last_time_update;
-
-                    localisation_robot.teammate = tracked_robot.teammate;
-                    localisation_robot.purpose  = tracked_robot.purpose;
-
-                    localisation_robots->robots.push_back(localisation_robot);
+
+                // Check if we have ground truth robot data from Webots
+                if (cfg.use_ground_truth && robots_ground_truth && !robots_ground_truth->robots.empty()) {
+                    log<DEBUG>("Using ground truth for localisation.");
+                    // Get our team colour as a string from GameState
+                    std::string our_team_colour =
+                        (game_state.team.team_colour == GameState::TeamColour::BLUE) ? "BLUE" : "RED";
+                    // Use ground truth data from Webots
+                    for (const auto& gt_robot : robots_ground_truth->robots) {
+                        // Skip our own robot using GlobalConfig player_id and team colour
+                        if (!(gt_robot.player_number == static_cast<int32_t>(PLAYER_ID)
```

Comment:
I wonder if its better to filter this in the webots module. Not a big deal though

Reply:
added as todo